doc/man7/flux-jobtap-plugins.rst

```diff
@@ -237,6 +237,21 @@ The callback should return 0 on success, and -1 on failure.  On failure,
 it may optionally set a human readable error string in the ``errstr`` output
 argument.  The ``flux_jobtap_error()`` convenience function may be useful here.
 
+
+PLUGIN CALLBACK TOPICS
+======================
+
+plugin.query
+  The job manager calls the ``plugin.query`` callback topic to give a plugin
+  the opportunity to provide extra data in response to a ``jobtap-query``
+  request (as used by the ``flux jobtap query PLUGIN`` command). This can be
+  used by a plugin to export internal plugin state for inspection by an admin
+  or user by placing the data in the output arguments of the callback, e.g.::
+
+    flux_plugin_arg_pack (p, FLUX_PLUGIN_ARG_OUT,
+                          "{s:o}"
+                          "data", internal_data);
```

Comment:
Since the args are destroyed after the callback returns, maybe the example should use `{s:O}` (big O)?

I'm not sure if that makes it any clearer.  This is a fertile ground for bugs though.

Reply:
Ah, very good point. Given my mistake above, it is probably always better to pack with "O" and decref manually. (I don't know how many times I have to relearn this until it gets through my head)